Atlas rotation

saarah815 opened the pull request:
## Description

**What is this PR**

- [ ] Bug fix
- [ ] Addition of a new feature
- [X] Other

**Why is this PR needed?**

Refactors atlas rotation code into a separate function for reuse.

**What does this PR do?**

No functionality change, just moved methods into new file in utils called `atlas_rotation.py`.

## References

#97 

## How has this PR been tested?

Runs in the same way as before. Tested with resetting rotation changes, tested with different samples, tested with changing all 3 rotation parameters.

## Is this a breaking change?

No

## Does this PR require an update to the documentation?

No

## Checklist:

- [X] The code has been tested locally
- [ ] Tests have been added to cover all new functionality (unit & integration)
- [N/A] The documentation has been updated to reflect any changes
- [X] The code has been formatted with [pre-commit](https://pre-commit.com/)

saarah815 commented:
Started working on #98 earlier today but didn't get round to creating a PR for #97 at the time - have just done this now.

saarah815 commented:
Sounds good, will do - thank you!

alessandrofelder reviewed (commented):
Thanks @saarah815 - this looks great. I think you've addressed all the comments so far.

My two-cents would be that it's good practice to add unit tests for the newly created functions in `transforms.py` as part of this PR, but I am equally happy for this to be done in a separate PR... what do you think? (Maybe there is a grand plan on how to approach testing that I missed?)

saarah815 commented:
> Thanks @saarah815 - this looks great. I think you've addressed all the comments so far.
> 
> My two-cents would be that it's good practice to add unit tests for the newly created functions in `transforms.py` as part of this PR, but I am equally happy for this to be done in a separate PR... what do you think? (Maybe there is a grand plan on how to approach testing that I missed?)

Sure, that sounds good. I can add the tests to this PR. Thank you!

alessandrofelder reviewed (approved):
Awesome, thanks!
